core/src/main/java/bisq/core/api/CoreOffersService.java

```diff
@@ -79,23 +84,28 @@ Offer getOffer(String id) {
                         new IllegalStateException(format("offer with id '%s' not found", id)));
     }
 
+    Offer getMyOffer(String id) {
+        return offerBookService.getOffers().stream()
+                .filter(o -> o.getId().equals(id))
+                .filter(o -> o.isMyOffer(keyRing))
+                .findAny().orElseThrow(() ->
+                        new IllegalStateException(format("offer with id '%s' not found", id)));
+    }
+
     List<Offer> getOffers(String direction, String currencyCode) {
         List<Offer> offers = offerBookService.getOffers().stream()
-                .filter(o -> {
-                    var offerOfWantedDirection = o.getDirection().name().equalsIgnoreCase(direction);
-                    var offerInWantedCurrency = o.getOfferPayload().getCounterCurrencyCode()
-                            .equalsIgnoreCase(currencyCode);
-                    return offerOfWantedDirection && offerInWantedCurrency;
-                })
+                .filter(o -> offerMatchesDirectionAndCurrency(o, direction, currencyCode))
                 .collect(Collectors.toList());
+        sortOffers(offers, direction);
+        return offers;
+    }
 
-        // A buyer probably wants to see sell orders in price ascending order.
-        // A seller probably wants to see buy orders in price descending order.
-        if (direction.equalsIgnoreCase(BUY.name()))
-            offers.sort(Comparator.comparing(Offer::getPrice).reversed());
-        else
-            offers.sort(Comparator.comparing(Offer::getPrice));
-
+    List<Offer> getMyOffers(String direction, String currencyCode) {
+        List<Offer> offers = offerBookService.getOffers().stream()
+                .filter(o -> o.isMyOffer(keyRing))
+                .filter(o -> offerMatchesDirectionAndCurrency(o, direction, currencyCode))
+                .collect(Collectors.toList());
+        sortOffers(offers, direction);
```

Comment:
Why not sort the offers in the stream, wouldn't that be more efficient? To make it more readable you could make a method `offerComparator`.

Reply:
Like this 9689c3edfcb04a20b97718c64a71d0aed56b9a4e ?

Reply:
Yeah. Since the stream is there anyway I'm thinking it would be possible to do some optimization by the stream interface, but I'm not sure. It's also easier to follow all the operations on Collection when lined up in stream operations. The Collection.sort() is probably more efficient as a stand alone sort method though.